### PR TITLE
Fix for Bug 196

### PR DESCRIPTION
**Pull request by amitla1**
Related: #196 

Added a try-catch statement in the OnFileFound method for the Docker component because component detection was attempting to scan files that are named like a Dockerfile but aren't really a Dockerfile which was causing an error.

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs [hunk 1]
```diff
@@ -36,19 +36,26 @@ public class DockerfileComponentDetector : FileComponentDetector, IDefaultOffCom
...
+                }
 
-            string contents;
-            using (var reader = new StreamReader(file.Stream))
+                var stageNameMap = new Dictionary<string, string>();
+                var dockerFileComponent = this.ParseDockerFile(contents, file.Location, singleFileComponentRecorder, stageNameMap);
+            } catch (Exception e)
             {
-                contents = await reader.ReadToEndAsync();
+                this.Logger.LogError($"The file is not a Dockerfile it just has 'Docker' in the name. \n Error Message: <{e.Message}>");
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs:56]: I would downgrade this to a warning. Error is usually used for unrecoverable issues. We are able to recover from this by skipping the file.

File: src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs [hunk 2]
```diff
@@ -36,19 +36,26 @@ public class DockerfileComponentDetector : FileComponentDetector, IDefaultOffCom
...
+                this.Logger.LogException(e, isError: true, printException: true);
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs:57]: I'm not sure we need the full exception printed here in quiet mode

```suggestion
                this.Logger.LogException(e);
```

**Review by JamieMagee — CHANGES_REQUESTED** [review]
Could you add a little more context on what the changes you made were and why they were made to the PR description. I know this is a small change, but it's a good habit to get into and can help us track down issues later.

File: src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs (hunk 1 above)
Comment (jcfiorenzano) [reply src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs:56]: What do you think of logging "The file doesn't appear to be a Dockerfile." instead?

Comment (JamieMagee) [reply src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs:56]: Yeah. Like this?

```suggestion
                this.Logger.LogError($"The file doesn't appear to be a Dockerfile: '{file.Location}'");
```

**Review by JamieMagee — APPROVED** [review]
Approved pending fix for indentation warning

> Elements should have the same indentation

**Pull request merged**